### PR TITLE
Fix: Returning a promise and calling context.done results in an invalid message

### LINKED ISSUE #49: Returning a promise and calling context.done results in an invalid message
If you export a promise instead of a function and call context.done you end up with the wrong error message: 
`Error: 'done' has already been called. Please check your script for extraneous calls to 'done'.`
Where the error should have read:
`Error: Choose either to return a promise or call 'done'.  Do not use both in your script.`
It would seem that the isPromise parameter in [Context.ts](https://github.com/Azure/azure-functions-nodejs-worker/blob/dev/src/Context.ts#L73) is never being set to `true` 
Which ends up causing the wrong message to be reported by the worker.
Also as far as I could tell the documentation is nowhere clear on what's the expected behavior when exporting promises and this may be confusing to users.

### PR DESCRIPTION
**Pull request by mhoeger**
Resolves #49

### CODE REVIEW

File: src/Context.ts [hunk 1]
```diff
@@ -72,6 +72,7 @@ class Context implements IContext {
     let _promise = false;
     // isPromise is a hidden parameter that we set to true in the event of a returned promise
     this.done = (err?: any, result?: any, isPromise?: boolean) => {
+      _promise = isPromise === true;
       if (_done) {
         if (_promise) {
           this.log.error("Error: Choose either to return a promise or call 'done'.  Do not use both in your script.");
```
[the remark this answers is not in the document]
Comment (mhoeger) [reply src/Context.ts]: Sortof! It'll at least make sure that the right error message is being logged..! are those logs coming from host?

[the remark this answers is not in the document]
Comment (pragnagopa) [reply src/Context.ts]: yes. NodejS worker writes them and logs show up system and user logs

File: src/WorkerChannel.ts [hunk 2]
```diff
@@ -148,7 +148,7 @@ export class WorkerChannel {
 
       if (result && isFunction(result.then)) {
         result.then(result => (<any>context.done)(null, result, true))
-          .catch(err => context.done(err));
+          .catch(err => (<any>context.done)(err, null, true));
```
[the remark this answers is not in the document]
Comment (mhoeger) [reply src/WorkerChannel.ts]: @pragnagopa - added tests. would you mind reviewing again?

File: test/ContextTests.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    it ('throws error on function calling context.done more than once', () => {
+        callbackTwice(_context);
+        sinon.assert.calledOnce(_logger);
+        expect(_logger);
+        sinon.assert.calledWith(_logger, rpc.RpcLog.Level.Error, "Error: 'done' has already been called. Please check your script for extraneous calls to 'done'.");
+    });
+})
+
+async function asyncThrowsError(context) {
```
Comment (pragnagopa) [inline test/ContextTests.ts]: Another test where context.done() is missing

**Pull request merged**